### ISSUE TITLE
MethodCountCheck with wrong tokens crashes Checkstyle

### ISSUE THREAD

**Issue by denizarsan** [checkstyle/checkstyle#4556]
$ javac C.java

$ cat C.java
```java
class C {
   void m() {}
}
```

$ cat config.xml
```
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="MethodCount">
      <property name="tokens" value="ENUM_DEF"/>
    </module>
  </module>
</module>
```

$ java -jar checkstyle-7.8.2-all.jar -c config.xml C.java
```
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing C.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:295)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:213)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:425)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:359)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:174)
Caused by: java.lang.NullPointerException
        at com.puppycrawl.tools.checkstyle.checks.sizes.MethodCountCheck$MethodCounter.access$000(MethodCountCheck.java:218)
        at com.puppycrawl.tools.checkstyle.checks.sizes.MethodCountCheck.raiseCounter(MethodCountCheck.java:140)
        at com.puppycrawl.tools.checkstyle.checks.sizes.MethodCountCheck.visitToken(MethodCountCheck.java:112)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:370)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:507)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:312)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:184)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:78)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:316)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:286)
        ... 4 more
Checkstyle ends with 1 errors.
```
---------------

The tool crashes with a NPE for the given config and class.  The problem is that the "tokens" property is missing CLASS_DEF in this case (and INTERFACE_DEF when that is the case).  While this can be viewed as a bug in the config, Checkstyle should gracefully report a warning instead of crashing with an uncaught NPE.

Please let me know if you want me to help with fixing this.

**Comment by rnveach** [comment]
@denizarsan The same here too. It is better to report using release versions.
We have had similar issues with this method in this release (#4534) so please confirm what commit your using.

Fee free to provide a fix if you can, but this might not be that simple.
We expect no warnings on config since the tokens are acceptable.
Just checking for null might not be enough. Ignoring a parent of the method will cause the method to be seen under a different class definition. So the method's parent has to be tracked for each method.

**Comment by denizarsan** [comment]
@rnveach I have updated my post. I'm using the latest release (7.8.2) from https://sourceforge.net/projects/checkstyle/files/checkstyle/

**Comment by denizarsan** [comment]
I agree fixing this is not as simple as it looked initially. I will defer looking into it until after I finish sending some more bug reports. 

Let me know if you want bug reports in some other format, e.g., PRs with failing unit tests if that's more convenient.

**Comment by romani** [comment]
> Let me know if you want bug reports in some other format, e.g., PRs with failing unit tests if that's more convenient.

thanks a lot for your reports, it is better to stay with format that we demand for issue.
We have strict policy on PR live period, we close PRs that incomplete and has no activity and give smb else right to finish fix or do it from scratch.

**Comment by rnveach** [comment]
@romani If we end up fixing this bug correctly, than it also includes the fixes for Issue #4539.
Is that ok or should we do a simple NPE fix here and a more complex fix in #4539?

**Comment by romani** [comment]
> Is that ok or should we do a simple NPE fix here and a more complex fix in #4539?

I prefer to fix Exceptions quicker and then thoroughly make Check more correct from validation perspective. Users are more affected by Exceptions rather than by false-positives.

**Comment by romani** [comment]
fix is merged
